### PR TITLE
Update Netty to 4.1.x (Java 17 compatibility)

### PR DESCRIPTION
**Pull request by roccodev**
This adds compatibility with newer versions of the JVM by updating Netty to the 4.1 branch.   
ProtocolLib >= 4.8 handles newer Netty versions in 1.8 well, and so should ViaVersion.

Docker images were also updated, though there's currently no official OpenJ9 image for JRE 17 + Alpine.

### CODE REVIEW

File: Spigot-Server-Patches/0239-Update-Netty-to-4.1.x.patch [hunk 1]
```diff
@@ -0,0 +1,453 @@
+From a6ee7874044ab88681f1c3d717e17660bc214e8a Mon Sep 17 00:00:00 2001
+From: RoccoDev <hey@rocco.dev>
+Date: Sat, 11 Mar 2023 14:33:39 +0100
+Subject: [PATCH] Update Netty to 4.1.x
+
+
+diff --git a/pom.xml b/pom.xml
```
Comment (archer-321) [inline Spigot-Server-Patches/0239-Update-Netty-to-4.1.x.patch:7]: I believe we overwrite this file in our CI with `/server_ci_pom.xml`. Could you please also update that file?

File: docker/j9.Dockerfile [hunk 2]
```diff
@@ -1,7 +1,8 @@
 ARG git_ref
 FROM ghcr.io/projectkig/kigpaper/kig-paper-builder:${git_ref} AS builder
 
-FROM ibmjava:8-jre-alpine
+# waiting for an official ibm-semeru-runtimes alpine image
+FROM cobaltinc:openj9-17.0.3_7-jre-alpine
```
Comment (archer-321) [inline docker/j9.Dockerfile]: Which container is this pulling? I found a container `cobaltinc/openj9-17.0.3_7-jre-alpine`, but the tag for that is the default `latest`.

Seeing as this container doesn't use tags and doesn't provide any information or source, I'm not comfortable using it in a production environment or releasing a Dockerfile that blindly trusts this image.
Are we going to use Hotspot until OpenJ9 on Alpine is released somewhere? If so, I believe we can just leave this Dockerfile broken for now and update once we have a reliable source. Alternatively, we could use a non-Alpine-based image, as we're not planning to use this anyway at the moment. This has the added benefit of not breaking our CI.

**Review by archer-321 — CHANGES_REQUESTED** [review]
I had two comments, but this PR looks good otherwise.
Thank you for updating this!

File: docker/j9.Dockerfile (hunk 2 above)
Comment (roccodev) [reply docker/j9.Dockerfile]: I agree, we can keep the J9 image on JRE 8 for now.

Ideally, we'd also want to use the Hotspot image just for building. We should build our own JRE image with `jlink`, but that should target the whole infrastructure, so it's not something for this repository.

**Review by archer-321 — APPROVED** [review]
LGTM. Thanks again!

**Pull request merged**